Misleading Stack Trace

As recommended by @brisvag, opening a new issue.  When loading a larger image volume, get the following error.  At first glance, it appears to be a vispy issue, but higher up in the stack trace it appears unrelated to vispy. He recommends making the errors more informative.

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/app/backends/_qt.py:903, in CanvasBackendDesktop.paintGL(self=<vispy.app.backends._qt.CanvasBackendDesktop object>)
    901 # (0, 0, self.width(), self.height()))
    902 self._vispy_canvas.set_current()
--> 903 self._vispy_canvas.events.draw(region=None)
        self._vispy_canvas = <VispyCanvas (PyQt5) at 0x2aacb7be2880>
        self._vispy_canvas.events.draw = <vispy.util.event.EventEmitter object at 0x2aacb7c3a7f0>
        self = <vispy.app.backends._qt.CanvasBackendDesktop object at 0x2aacb7c460d0>
        self._vispy_canvas.events = <vispy.util.event.EmitterGroup object at 0x2aacb7c3a670>
    905 # Clear the alpha channel with QOpenGLWidget (Qt >= 5.4), otherwise the
    906 # window is translucent behind non-opaque objects.
    907 # Reference:  MRtrix3/mrtrix3#266
    908 if QT5_NEW_API or PYSIDE6_API or PYQT6_API:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:453, in EventEmitter.__call__(self=<vispy.util.event.EventEmitter object>, *args=(), **kwargs={'region': None})
    450 if self._emitting > 1:
    451     raise RuntimeError('EventEmitter loop detected!')
--> 453 self._invoke_callback(cb, event)
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        self = <vispy.util.event.EventEmitter object at 0x2aacb7c3a7f0>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb7be2880>>
    454 if event.blocked:
    455     break

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:471, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    469     cb(event)
    470 except Exception:
--> 471     _handle_exception(self.ignore_callback_errors,
        self = <vispy.util.event.EventEmitter object at 0x2aacb7c3a7f0>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb7be2880>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        (cb, event) = (<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb7be2880>>, <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    472                       self.print_callback_errors,
    473                       self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:469, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    467 def _invoke_callback(self, cb, event):
    468     try:
--> 469         cb(event)
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb7be2880>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
    470     except Exception:
    471         _handle_exception(self.ignore_callback_errors,
    472                           self.print_callback_errors,
    473                           self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:218, in SceneCanvas.on_draw(self=<VispyCanvas (PyQt5)>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    215 # Now that a draw event is going to be handled, open up the
    216 # scheduling of further updates
    217 self._update_pending = False
--> 218 self._draw_scene()
        self = <VispyCanvas (PyQt5) at 0x2aacb7be2880>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:277, in SceneCanvas._draw_scene(self=<VispyCanvas (PyQt5)>, bgcolor=<class 'numpy.ndarray'> (4,) float32)
    275     bgcolor = self._bgcolor
    276 self.context.clear(color=bgcolor, depth=True)
--> 277 self.draw_visual(self.scene)
        self = <VispyCanvas (PyQt5) at 0x2aacb7be2880>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:315, in SceneCanvas.draw_visual(self=<VispyCanvas (PyQt5)>, visual=<SubScene>, event=None)
    313         else:
    314             if hasattr(node, 'draw'):
--> 315                 node.draw()
        node = <Volume at 0x2aacbc72afa0>
    316                 prof.mark(str(node))
    317 else:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/visuals.py:103, in VisualNode.draw(self=<Volume>)
    101 if self.picking and not self.interactive:
    102     return
--> 103 self._visual_superclass.draw(self)
        self = <Volume at 0x2aacbc72afa0>
        self._visual_superclass = <class 'vispy.visuals.volume.VolumeVisual'>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/visual.py:451, in Visual.draw(self=<Volume>)
    449 self._configure_gl_state()
    450 try:
--> 451     self._program.draw(self._vshare.draw_mode,
        self._vshare.draw_mode = 'triangle_strip'
        self = <Volume at 0x2aacbc72afa0>
        self._vshare = <vispy.visuals.visual.VisualShare object at 0x2aacbc492820>
        self._program = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbb770670>
        self._vshare.index_buffer = <IndexBuffer size=14 last_dim=1>
    452                        self._vshare.index_buffer)
    453 except Exception:
    454     logger.warning("Error drawing visual %r" % self)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/shaders/program.py:102, in ModularProgram.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, *args=('triangle_strip', <IndexBuffer size=14 last_dim=1>), **kwargs={})
    100 self.build_if_needed()
    101 self.update_variables()
--> 102 Program.draw(self, *args, **kwargs)
        self = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbb770670>
        args = ('triangle_strip', <IndexBuffer size=14 last_dim=1>)
        kwargs = {}

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/program.py:526, in Program.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, mode='triangle_strip', indices=<IndexBuffer size=14 last_dim=1>, check_error=True)
    522     raise TypeError("Invalid index: %r (must be IndexBuffer)" %
    523                     indices)
    525 # Process GLIR commands
--> 526 canvas.context.flush_commands()
        canvas = <VispyCanvas (PyQt5) at 0x2aacb7be2880>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/context.py:172, in GLContext.flush_commands(self=<GLContext>, event=None)
    170         fbo = 0
    171     self.shared.parser.parse([('CURRENT', 0, fbo)])
--> 172 self.glir.flush(self.shared.parser)
        self = <GLContext at 0x2aacb7c3a7c0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:582, in GlirQueue.flush(self=<vispy.gloo.glir.GlirQueue object>, parser=<vispy.gloo.glir.GlirParser object>)
    580 def flush(self, parser):
    581     """Flush all current commands to the GLIR interpreter."""
--> 582     self._shared.flush(parser)
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7c3a940>
        self._shared = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7c3aa00>
        self = <vispy.gloo.glir.GlirQueue object at 0x2aacb7c3a9d0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:504, in _GlirQueueShare.flush(self=<vispy.gloo.glir._GlirQueueShare object>, parser=<vispy.gloo.glir.GlirParser object>)
    502     show = self._verbose if isinstance(self._verbose, str) else None
    503     self.show(show)
--> 504 parser.parse(self._filter(self.clear(), parser))
        self = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7c3aa00>
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7c3a940>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:822, in GlirParser.parse(self=<vispy.gloo.glir.GlirParser object>, commands=[('FUNC', 'glClearColor', 0.0, 0.0, 0.0, 1.0), ('FUNC', 'glClear', 17664), ('SIZE', 79, 96), ('DATA', 79, 0, <class 'numpy.ndarray'> (8, 1) [('f0', '<f4', (3,))]), ('SIZE', 82, 56), ('DATA', 82, 0, <class 'numpy.ndarray'> (14,) uint32), ('FUNC', 'glDisable', 'cull_face'), ('FUNC', 'glEnable', 'blend'), ('FUNC', 'glEnable', 'depth_test'), ('FUNC', 'glBlendFuncSeparate', 'src_alpha', 'one_minus_src_alpha', 'zero', 'one'), ('DRAW', 81, 'triangle_strip', (82, 'UNSIGNED_INT', 14))])
    819     self._objects.pop(id_)
    821 for command in commands:
--> 822     self._parse(command)
        command = ('SIZE', 79, 96)
        self = <vispy.gloo.glir.GlirParser object at 0x2aacb7c3a940>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:778, in GlirParser._parse(self=<vispy.gloo.glir.GlirParser object>, command=('SIZE', 79, 96))
    776 if ob is None:
    777     if id_ not in self._invalid_objects:
--> 778         raise RuntimeError('Cannot %s object %i because it '
        'Cannot %s object %i because it '
                                       'does not exist' % (cmd, id_) = 'Cannot SIZE object 79 because it does not exist'
        cmd = 'SIZE'
        id_ = 79
        (cmd, id_) = ('SIZE', 79)
    779                            'does not exist' % (cmd, id_))
    780     return
    781 # Triage over command. Order of commands is set so most
    782 # common ones occur first.

RuntimeError: Cannot SIZE object 79 because it does not exist
```

The output in my terminal is as follows (which includes the automatic update process and error):

```
[kdean@NucleusC052 ~]$ '/archive/MIL/dean/applications/napari.sh' 
Requirement already satisfied: pip in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (22.1.1)
Requirement already satisfied: napari in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (0.4.16rc3)
Requirement already satisfied: toolz>=0.10.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.11.2)
Requirement already satisfied: scikit-image>=0.19.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.19.2)
Requirement already satisfied: certifi>=2018.1.18 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2021.10.8)
Requirement already satisfied: numpy>=1.18.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.22.3)
Requirement already satisfied: PyYAML>=5.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (6.0)
Requirement already satisfied: appdirs>=1.4.4 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.4.4)
Requirement already satisfied: pandas>=1.3.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.4.2)
Requirement already satisfied: jsonschema>=3.2.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (4.4.0)
Requirement already satisfied: npe2>=0.3.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.3.0)
Requirement already satisfied: typing-extensions in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (4.2.0)
Requirement already satisfied: tifffile>=2020.2.16 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2022.4.22)
Requirement already satisfied: cachey>=0.2.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.2.1)
Requirement already satisfied: dask[array]!=2.28.0,>=2.15.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2022.4.1)
Requirement already satisfied: napari-console>=0.0.4 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.0.4)
Requirement already satisfied: vispy>=0.10.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.10.0)
Requirement already satisfied: napari-svg>=0.1.4 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.1.6)
Requirement already satisfied: pint>=0.17 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.19.2)
Requirement already satisfied: wrapt>=1.11.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.14.0)
Requirement already satisfied: napari-plugin-engine>=0.1.9 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.2.0)
Requirement already satisfied: pygments>=2.4.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2.12.0)
Requirement already satisfied: PyOpenGL>=3.1.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (3.1.6)
Requirement already satisfied: psutil>=5.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (5.9.0)
Requirement already satisfied: tqdm>=4.56.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (4.64.0)
Requirement already satisfied: Pillow!=7.1.0,!=7.1.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (9.1.0)
Requirement already satisfied: magicgui>=0.3.6 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.4.0)
Requirement already satisfied: superqt>=0.2.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (0.3.1)
Requirement already satisfied: qtpy>=1.7.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2.0.1)
Requirement already satisfied: numpydoc>=0.9.2 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.2.1)
Requirement already satisfied: scipy>=1.5.4 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.8.0)
Requirement already satisfied: pydantic>=1.9.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (1.9.0)
Requirement already satisfied: imageio!=2.11.0,>=2.5.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari) (2.18.0)
Requirement already satisfied: heapdict in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from cachey>=0.2.1->napari) (1.0.1)
Requirement already satisfied: cloudpickle>=1.1.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from dask[array]!=2.28.0,>=2.15.0->napari) (2.0.0)
Requirement already satisfied: fsspec>=0.6.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from dask[array]!=2.28.0,>=2.15.0->napari) (2022.3.0)
Requirement already satisfied: packaging>=20.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from dask[array]!=2.28.0,>=2.15.0->napari) (21.3)
Requirement already satisfied: partd>=0.3.10 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from dask[array]!=2.28.0,>=2.15.0->napari) (1.2.0)
Requirement already satisfied: attrs>=17.4.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from jsonschema>=3.2.0->napari) (21.4.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from jsonschema>=3.2.0->napari) (0.18.1)
Requirement already satisfied: psygnal>=0.3.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from magicgui>=0.3.6->napari) (0.3.3)
Requirement already satisfied: docstring-parser in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from magicgui>=0.3.6->napari) (0.14)
Requirement already satisfied: ipykernel>=5.2.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari-console>=0.0.4->napari) (6.13.0)
Requirement already satisfied: qtconsole!=4.7.6,>=4.5.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari-console>=0.0.4->napari) (5.3.0)
Requirement already satisfied: IPython>=7.7.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from napari-console>=0.0.4->napari) (8.2.0)
Requirement already satisfied: pytomlpp in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from npe2>=0.3.0->napari) (1.0.11)
Requirement already satisfied: typer in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from npe2>=0.3.0->napari) (0.4.1)
Requirement already satisfied: Jinja2<3.1,>=2.10 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from numpydoc>=0.9.2->napari) (3.0.3)
Requirement already satisfied: sphinx>=1.8 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from numpydoc>=0.9.2->napari) (4.5.0)
Requirement already satisfied: pytz>=2020.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from pandas>=1.3.0->napari) (2022.1)
Requirement already satisfied: python-dateutil>=2.8.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from pandas>=1.3.0->napari) (2.8.2)
Requirement already satisfied: networkx>=2.2 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from scikit-image>=0.19.1->napari) (2.8)
Requirement already satisfied: PyWavelets>=1.1.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from scikit-image>=0.19.1->napari) (1.3.0)
Requirement already satisfied: hsluv in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from vispy>=0.10.0->napari) (5.0.2)
Requirement already satisfied: freetype-py in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from vispy>=0.10.0->napari) (2.3.0)
Requirement already satisfied: kiwisolver in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from vispy>=0.10.0->napari) (1.4.2)
Requirement already satisfied: debugpy>=1.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (1.6.0)
Requirement already satisfied: tornado>=6.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (6.1)
Requirement already satisfied: nest-asyncio in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (1.5.5)
Requirement already satisfied: matplotlib-inline>=0.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (0.1.3)
Requirement already satisfied: jupyter-client>=6.1.12 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (7.3.0)
Requirement already satisfied: traitlets>=5.1.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from ipykernel>=5.2.0->napari-console>=0.0.4->napari) (5.1.1)
Requirement already satisfied: jedi>=0.16 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (0.18.1)
Requirement already satisfied: decorator in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (5.1.1)
Requirement already satisfied: pickleshare in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (0.7.5)
Requirement already satisfied: stack-data in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (0.2.0)
Requirement already satisfied: setuptools>=18.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (62.1.0)
Requirement already satisfied: pexpect>4.3 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (4.8.0)
Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (3.0.29)
Requirement already satisfied: backcall in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from IPython>=7.7.0->napari-console>=0.0.4->napari) (0.2.0)
Requirement already satisfied: MarkupSafe>=2.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from Jinja2<3.1,>=2.10->numpydoc>=0.9.2->napari) (2.1.1)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from packaging>=20.0->dask[array]!=2.28.0,>=2.15.0->napari) (3.0.8)
Requirement already satisfied: locket in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from partd>=0.3.10->dask[array]!=2.28.0,>=2.15.0->napari) (1.0.0)
Requirement already satisfied: six>=1.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas>=1.3.0->napari) (1.16.0)
Requirement already satisfied: jupyter-core in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from qtconsole!=4.7.6,>=4.5.1->napari-console>=0.0.4->napari) (4.10.0)
Requirement already satisfied: pyzmq>=17.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from qtconsole!=4.7.6,>=4.5.1->napari-console>=0.0.4->napari) (22.3.0)
Requirement already satisfied: ipython-genutils in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from qtconsole!=4.7.6,>=4.5.1->napari-console>=0.0.4->napari) (0.2.0)
Requirement already satisfied: imagesize in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.3.0)
Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (2.0.0)
Requirement already satisfied: sphinxcontrib-applehelp in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.0.2)
Requirement already satisfied: sphinxcontrib-jsmath in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.0.1)
Requirement already satisfied: sphinxcontrib-devhelp in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.0.2)
Requirement already satisfied: sphinxcontrib-qthelp in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.0.3)
Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (1.1.5)
Requirement already satisfied: docutils<0.18,>=0.14 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (0.17.1)
Requirement already satisfied: snowballstemmer>=1.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (2.2.0)
Requirement already satisfied: alabaster<0.8,>=0.7 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (0.7.12)
Requirement already satisfied: babel>=1.3 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (2.10.1)
Requirement already satisfied: importlib-metadata>=4.4 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (4.11.3)
Requirement already satisfied: requests>=2.5.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from sphinx>=1.8->numpydoc>=0.9.2->napari) (2.27.1)
Requirement already satisfied: click<9.0.0,>=7.1.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from typer->npe2>=0.3.0->napari) (8.1.2)
Requirement already satisfied: zipp>=0.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from importlib-metadata>=4.4->sphinx>=1.8->numpydoc>=0.9.2->napari) (3.8.0)
Requirement already satisfied: parso<0.9.0,>=0.8.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from jedi>=0.16->IPython>=7.7.0->napari-console>=0.0.4->napari) (0.8.3)
Requirement already satisfied: entrypoints in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from jupyter-client>=6.1.12->ipykernel>=5.2.0->napari-console>=0.0.4->napari) (0.4)
Requirement already satisfied: ptyprocess>=0.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from pexpect>4.3->IPython>=7.7.0->napari-console>=0.0.4->napari) (0.7.0)
Requirement already satisfied: wcwidth in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->IPython>=7.7.0->napari-console>=0.0.4->napari) (0.2.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from requests>=2.5.0->sphinx>=1.8->numpydoc>=0.9.2->napari) (1.26.9)
Requirement already satisfied: idna<4,>=2.5 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from requests>=2.5.0->sphinx>=1.8->numpydoc>=0.9.2->napari) (3.3)
Requirement already satisfied: charset-normalizer~=2.0.0 in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from requests>=2.5.0->sphinx>=1.8->numpydoc>=0.9.2->napari) (2.0.12)
Requirement already satisfied: pure-eval in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from stack-data->IPython>=7.7.0->napari-console>=0.0.4->napari) (0.2.2)
Requirement already satisfied: asttokens in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from stack-data->IPython>=7.7.0->napari-console>=0.0.4->napari) (2.0.5)
Requirement already satisfied: executing in /endosome/archive/MIL/dean/.conda/envs/napari-env/lib/python3.9/site-packages (from stack-data->IPython>=7.7.0->napari-console>=0.0.4->napari) (0.8.3)
WARNING: Error drawing visual <Volume at 0x2aacb86b0640>
11:54:43 WARNING Error drawing visual <Volume at 0x2aacb86b0640>
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/latebind.py:43, in LateBind.__call__(self=<OpenGL.platform.baseplatform.glTexImage3D object>, *args=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None), **named={})
     42 try:
---> 43     return self._finalCall( *args, **named )
        self._finalCall = <function Wrapper.finaliseCall.<locals>.wrapperCall at 0x2aacbd32e280>
        self = <OpenGL.platform.baseplatform.glTexImage3D object at 0x2aab14a4c8c0>
        args = (GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None)
        named = {}
     44 except (TypeError,AttributeError) as err:

TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

GLError                                   Traceback (most recent call last)
File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/app/backends/_qt.py:903, in CanvasBackendDesktop.paintGL(self=<vispy.app.backends._qt.CanvasBackendDesktop object>)
    901 # (0, 0, self.width(), self.height()))
    902 self._vispy_canvas.set_current()
--> 903 self._vispy_canvas.events.draw(region=None)
        self._vispy_canvas = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>
        self._vispy_canvas.events.draw = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        self = <vispy.app.backends._qt.CanvasBackendDesktop object at 0x2aacb7a55310>
        self._vispy_canvas.events = <vispy.util.event.EmitterGroup object at 0x2aacb7a518e0>
    905 # Clear the alpha channel with QOpenGLWidget (Qt >= 5.4), otherwise the
    906 # window is translucent behind non-opaque objects.
    907 # Reference:  MRtrix3/mrtrix3#266
    908 if QT5_NEW_API or PYSIDE6_API or PYQT6_API:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:453, in EventEmitter.__call__(self=<vispy.util.event.EventEmitter object>, *args=(), **kwargs={'region': None})
    450 if self._emitting > 1:
    451     raise RuntimeError('EventEmitter loop detected!')
--> 453 self._invoke_callback(cb, event)
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        self = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
    454 if event.blocked:
    455     break

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:471, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    469     cb(event)
    470 except Exception:
--> 471     _handle_exception(self.ignore_callback_errors,
        self = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        (cb, event) = (<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>, <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    472                       self.print_callback_errors,
    473                       self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:469, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    467 def _invoke_callback(self, cb, event):
    468     try:
--> 469         cb(event)
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
    470     except Exception:
    471         _handle_exception(self.ignore_callback_errors,
    472                           self.print_callback_errors,
    473                           self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:218, in SceneCanvas.on_draw(self=<VispyCanvas (PyQt5)>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    215 # Now that a draw event is going to be handled, open up the
    216 # scheduling of further updates
    217 self._update_pending = False
--> 218 self._draw_scene()
        self = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:277, in SceneCanvas._draw_scene(self=<VispyCanvas (PyQt5)>, bgcolor=<class 'numpy.ndarray'> (4,) float32)
    275     bgcolor = self._bgcolor
    276 self.context.clear(color=bgcolor, depth=True)
--> 277 self.draw_visual(self.scene)
        self = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:315, in SceneCanvas.draw_visual(self=<VispyCanvas (PyQt5)>, visual=<SubScene>, event=None)
    313         else:
    314             if hasattr(node, 'draw'):
--> 315                 node.draw()
        node = <Volume at 0x2aacb86b0640>
    316                 prof.mark(str(node))
    317 else:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/visuals.py:103, in VisualNode.draw(self=<Volume>)
    101 if self.picking and not self.interactive:
    102     return
--> 103 self._visual_superclass.draw(self)
        self = <Volume at 0x2aacb86b0640>
        self._visual_superclass = <class 'vispy.visuals.volume.VolumeVisual'>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/visual.py:451, in Visual.draw(self=<Volume>)
    449 self._configure_gl_state()
    450 try:
--> 451     self._program.draw(self._vshare.draw_mode,
        self._vshare.draw_mode = 'triangle_strip'
        self = <Volume at 0x2aacb86b0640>
        self._vshare = <vispy.visuals.visual.VisualShare object at 0x2aacbd3239d0>
        self._program = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbd3427f0>
        self._vshare.index_buffer = <IndexBuffer size=14 last_dim=1>
    452                        self._vshare.index_buffer)
    453 except Exception:
    454     logger.warning("Error drawing visual %r" % self)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/shaders/program.py:102, in ModularProgram.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, *args=('triangle_strip', <IndexBuffer size=14 last_dim=1>), **kwargs={})
    100 self.build_if_needed()
    101 self.update_variables()
--> 102 Program.draw(self, *args, **kwargs)
        self = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbd3427f0>
        args = ('triangle_strip', <IndexBuffer size=14 last_dim=1>)
        kwargs = {}

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/program.py:526, in Program.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, mode='triangle_strip', indices=<IndexBuffer size=14 last_dim=1>, check_error=True)
    522     raise TypeError("Invalid index: %r (must be IndexBuffer)" %
    523                     indices)
    525 # Process GLIR commands
--> 526 canvas.context.flush_commands()
        canvas = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/context.py:172, in GLContext.flush_commands(self=<GLContext>, event=None)
    170         fbo = 0
    171     self.shared.parser.parse([('CURRENT', 0, fbo)])
--> 172 self.glir.flush(self.shared.parser)
        self = <GLContext at 0x2aacb7a51a30>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:582, in GlirQueue.flush(self=<vispy.gloo.glir.GlirQueue object>, parser=<vispy.gloo.glir.GlirParser object>)
    580 def flush(self, parser):
    581     """Flush all current commands to the GLIR interpreter."""
--> 582     self._shared.flush(parser)
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>
        self._shared = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7a51c70>
        self = <vispy.gloo.glir.GlirQueue object at 0x2aacb7a51c40>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:504, in _GlirQueueShare.flush(self=<vispy.gloo.glir._GlirQueueShare object>, parser=<vispy.gloo.glir.GlirParser object>)
    502     show = self._verbose if isinstance(self._verbose, str) else None
    503     self.show(show)
--> 504 parser.parse(self._filter(self.clear(), parser))
        self = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7a51c70>
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:822, in GlirParser.parse(self=<vispy.gloo.glir.GlirParser object>, commands=[('FUNC', 'glViewport', 0, 0, 943, 776), ('FUNC', 'glClearColor', 0.0, 0.0, 0.0, 1.0), ('FUNC', 'glClear', 17664), ('FUNC', 'glDisable', 'cull_face'), ('FUNC', 'glDisable', 'depth_test'), ('FUNC', 'glEnable', 'blend'), ('FUNC', 'glBlendFuncSeparate', 'src_alpha', 'one_minus_src_alpha', 'zero', 'one'), ('CREATE', 41, 'Program'), ('CREATE', 52, 'VertexShader'), ('DATA', 52, 0, 'uniform mat4 u_matrix;\n\nvec4 affine_transform_ma...ransform_map_chain_11(vec4(v_position, 1.0));\n}\n\n'), ('ATTACH', 41, 52), ('CREATE', 53, 'FragmentShader'), ('DATA', 53, 0, 'uniform mat4 u_matrix;\n\nvec4 affine_transform_ma...)/2.0;\n\n    apply_alpha();\npicking_filter();\n\n}\n\n'), ('ATTACH', 41, 53), ('LINK', 41), ('DELETE', 52), ('DELETE', 53), ('UNIFORM', 41, 'u_alpha', 'float', <class 'numpy.ndarray'> (1,) float32), ('UNIFORM', 41, 'u_enabled', 'int', <class 'numpy.ndarray'> (1,) int32), ('UNIFORM', 41, 'u_id_color', 'vec4', <class 'numpy.ndarray'> (4,) float32), ...])
    819     self._objects.pop(id_)
    821 for command in commands:
--> 822     self._parse(command)
        command = ('SIZE', 40, (750, 2048, 2048, 1), 'luminance', None)
        self = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:794, in GlirParser._parse(self=<vispy.gloo.glir.GlirParser object>, command=('SIZE', 40, (750, 2048, 2048, 1), 'luminance', None))
    792     ob.set_data(*args)
    793 elif cmd == 'SIZE':  # VertexBuffer, IndexBuffer,
--> 794     ob.set_size(*args)  # Texture[1D, 2D, 3D], RenderBuffer
        args = ((750, 2048, 2048, 1), 'luminance', None)
        ob = <GlirTexture3D 40 at 0x2aacbd34a8e0>
    795 elif cmd == 'ATTACH':  # FrameBuffer, Program
    796     ob.attach(*args)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:1631, in GlirTexture3D.set_size(self=<GlirTexture3D 40>, shape=(750, 2048, 2048, 1), format=GL_LUMINANCE, internalformat=GL_LUMINANCE)
   1629 self.activate()
   1630 self._shape_formats = shape, format, internalformat
-> 1631 glTexImage3D(self._target, 0, internalformat, format,
        format = GL_LUMINANCE
        internalformat = GL_LUMINANCE
        self = <GlirTexture3D 40 at 0x2aacbd34a8e0>
        self._target = GL_TEXTURE_3D
        shape = (750, 2048, 2048, 1)
        gl.GL_BYTE = GL_BYTE
        shape[:3] = (750, 2048, 2048)
        gl = <module 'vispy.gloo.gl' from '/home2/kdean/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/gl/__init__.py'>
   1632              gl.GL_BYTE, shape[:3])

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:1580, in glTexImage3D(target=GL_TEXTURE_3D, level=0, internalformat=GL_LUMINANCE, format=GL_LUMINANCE, type=GL_BYTE, pixels=(750, 2048, 2048))
   1578 assert isinstance(pixels, (tuple, list))  # the only way we use this now
   1579 depth, height, width = pixels
-> 1580 _gl.glTexImage3D(target, level, internalformat,
        _gl = <module 'OpenGL.GL' from '/home2/kdean/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/GL/__init__.py'>
        _gl.glTexImage3D = <OpenGL.platform.baseplatform.glTexImage3D object at 0x2aab14a4c8c0>
        target = GL_TEXTURE_3D
        level = 0
        internalformat = GL_LUMINANCE
        border = 0
        depth = 750
        height = 2048
        width = 2048
        format = GL_LUMINANCE
        type = GL_BYTE
   1581                  width, height, depth, border, format, type, None)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/latebind.py:47, in LateBind.__call__(self=<OpenGL.platform.baseplatform.glTexImage3D object>, *args=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None), **named={})
     45 if self._finalCall is None:
     46     self._finalCall = self.finalise()
---> 47 return self._finalCall( *args, **named )
        self._finalCall = <function Wrapper.finaliseCall.<locals>.wrapperCall at 0x2aacbd32e280>
        self = <OpenGL.platform.baseplatform.glTexImage3D object at 0x2aab14a4c8c0>
        args = (GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None)
        named = {}

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/wrapper.py:889, in Wrapper.finaliseCall.<locals>.wrapperCall(*args=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None))
    887     err.cArgs = cArguments
    888     err.pyArgs = pyArgs
--> 889     raise err
    890 return result

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/wrapper.py:882, in Wrapper.finaliseCall.<locals>.wrapperCall(*args=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None))
    880 cArguments = pyArgs
    881 try:
--> 882     result = wrappedOperation( *cArguments )
        cArguments = (GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None)
        wrappedOperation = <OpenGL.platform.baseplatform.glTexImage3D object at 0x2aab107df970>
    883 except ctypes.ArgumentError as err:
    884     err.args = err.args + (cArguments,)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/platform/baseplatform.py:415, in _NullFunctionPointer.__call__(self=<OpenGL.platform.baseplatform.glTexImage3D object>, *args=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None), **named={})
    413 def __call__( self, *args, **named ):
    414     if self.load():
--> 415         return self( *args, **named )
        self = <OpenGL.platform.baseplatform.glTexImage3D object at 0x2aab107df970>
        args = (GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None)
        named = {}
    416     else:
    417         try:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/OpenGL/error.py:230, in _ErrorChecker.glCheckError(self=<OpenGL.error._ErrorChecker object>, result=None, baseOperation=<CFunctionType object>, cArguments=(GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None), *args=())
    228 err = self._currentChecker()
    229 if err != self._noErrorResult:
--> 230     raise self._errorClass(
        self = <OpenGL.error._ErrorChecker object at 0x2aab1071a460>
        self._errorClass = <class 'OpenGL.error.GLError'>
        err = 1285
        result = None
        cArguments = (GL_TEXTURE_3D, 0, GL_LUMINANCE, 2048, 2048, 750, 0, GL_LUMINANCE, GL_BYTE, None)
        baseOperation = <CFunctionType object at 0x2aab16c15ac0>
    231         err,
    232         result,
    233         cArguments = cArguments,
    234         baseOperation = baseOperation,
    235     )
    236 return result

GLError: GLError(
	err = 1285,
	description = b'out of memory',
	baseOperation = glTexImage3D,
	pyArgs = (
		GL_TEXTURE_3D,
		0,
		GL_LUMINANCE,
		2048,
		2048,
		750,
		0,
		GL_LUMINANCE,
		GL_BYTE,
		None,
	),
	cArgs = (
		GL_TEXTURE_3D,
		0,
		GL_LUMINANCE,
		2048,
		2048,
		750,
		0,
		GL_LUMINANCE,
		GL_BYTE,
		None,
	),
	cArguments = (
		GL_TEXTURE_3D,
		0,
		GL_LUMINANCE,
		2048,
		2048,
		750,
		0,
		GL_LUMINANCE,
		GL_BYTE,
		None,
	)
)
WARNING: Error drawing visual <Volume at 0x2aacb86b0640>
11:54:45 WARNING Error drawing visual <Volume at 0x2aacb86b0640>
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/app/backends/_qt.py:903, in CanvasBackendDesktop.paintGL(self=<vispy.app.backends._qt.CanvasBackendDesktop object>)
    901 # (0, 0, self.width(), self.height()))
    902 self._vispy_canvas.set_current()
--> 903 self._vispy_canvas.events.draw(region=None)
        self._vispy_canvas = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>
        self._vispy_canvas.events.draw = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        self = <vispy.app.backends._qt.CanvasBackendDesktop object at 0x2aacb7a55310>
        self._vispy_canvas.events = <vispy.util.event.EmitterGroup object at 0x2aacb7a518e0>
    905 # Clear the alpha channel with QOpenGLWidget (Qt >= 5.4), otherwise the
    906 # window is translucent behind non-opaque objects.
    907 # Reference:  MRtrix3/mrtrix3#266
    908 if QT5_NEW_API or PYSIDE6_API or PYQT6_API:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:453, in EventEmitter.__call__(self=<vispy.util.event.EventEmitter object>, *args=(), **kwargs={'region': None})
    450 if self._emitting > 1:
    451     raise RuntimeError('EventEmitter loop detected!')
--> 453 self._invoke_callback(cb, event)
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        self = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
    454 if event.blocked:
    455     break

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:471, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    469     cb(event)
    470 except Exception:
--> 471     _handle_exception(self.ignore_callback_errors,
        self = <vispy.util.event.EventEmitter object at 0x2aacb7a51a60>
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
        (cb, event) = (<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>, <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    472                       self.print_callback_errors,
    473                       self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/util/event.py:469, in EventEmitter._invoke_callback(self=<vispy.util.event.EventEmitter object>, cb=<bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5)>>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    467 def _invoke_callback(self, cb, event):
    468     try:
--> 469         cb(event)
        cb = <bound method SceneCanvas.on_draw of <VispyCanvas (PyQt5) at 0x2aacb79ffb50>>
        event = <DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>
    470     except Exception:
    471         _handle_exception(self.ignore_callback_errors,
    472                           self.print_callback_errors,
    473                           self, cb_event=(cb, event))

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:218, in SceneCanvas.on_draw(self=<VispyCanvas (PyQt5)>, event=<DrawEvent blocked=False handled=False native=None region=None source=None sources=[] type=draw>)
    215 # Now that a draw event is going to be handled, open up the
    216 # scheduling of further updates
    217 self._update_pending = False
--> 218 self._draw_scene()
        self = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:277, in SceneCanvas._draw_scene(self=<VispyCanvas (PyQt5)>, bgcolor=<class 'numpy.ndarray'> (4,) float32)
    275     bgcolor = self._bgcolor
    276 self.context.clear(color=bgcolor, depth=True)
--> 277 self.draw_visual(self.scene)
        self = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/canvas.py:315, in SceneCanvas.draw_visual(self=<VispyCanvas (PyQt5)>, visual=<SubScene>, event=None)
    313         else:
    314             if hasattr(node, 'draw'):
--> 315                 node.draw()
        node = <Volume at 0x2aacb86b0640>
    316                 prof.mark(str(node))
    317 else:

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/scene/visuals.py:103, in VisualNode.draw(self=<Volume>)
    101 if self.picking and not self.interactive:
    102     return
--> 103 self._visual_superclass.draw(self)
        self = <Volume at 0x2aacb86b0640>
        self._visual_superclass = <class 'vispy.visuals.volume.VolumeVisual'>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/visual.py:451, in Visual.draw(self=<Volume>)
    449 self._configure_gl_state()
    450 try:
--> 451     self._program.draw(self._vshare.draw_mode,
        self._vshare.draw_mode = 'triangle_strip'
        self = <Volume at 0x2aacb86b0640>
        self._vshare = <vispy.visuals.visual.VisualShare object at 0x2aacbd3239d0>
        self._program = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbd3427f0>
        self._vshare.index_buffer = <IndexBuffer size=14 last_dim=1>
    452                        self._vshare.index_buffer)
    453 except Exception:
    454     logger.warning("Error drawing visual %r" % self)

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/visuals/shaders/program.py:102, in ModularProgram.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, *args=('triangle_strip', <IndexBuffer size=14 last_dim=1>), **kwargs={})
    100 self.build_if_needed()
    101 self.update_variables()
--> 102 Program.draw(self, *args, **kwargs)
        self = <vispy.visuals.shaders.program.ModularProgram object at 0x2aacbd3427f0>
        args = ('triangle_strip', <IndexBuffer size=14 last_dim=1>)
        kwargs = {}

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/program.py:526, in Program.draw(self=<vispy.visuals.shaders.program.ModularProgram object>, mode='triangle_strip', indices=<IndexBuffer size=14 last_dim=1>, check_error=True)
    522     raise TypeError("Invalid index: %r (must be IndexBuffer)" %
    523                     indices)
    525 # Process GLIR commands
--> 526 canvas.context.flush_commands()
        canvas = <VispyCanvas (PyQt5) at 0x2aacb79ffb50>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/context.py:172, in GLContext.flush_commands(self=<GLContext>, event=None)
    170         fbo = 0
    171     self.shared.parser.parse([('CURRENT', 0, fbo)])
--> 172 self.glir.flush(self.shared.parser)
        self = <GLContext at 0x2aacb7a51a30>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:582, in GlirQueue.flush(self=<vispy.gloo.glir.GlirQueue object>, parser=<vispy.gloo.glir.GlirParser object>)
    580 def flush(self, parser):
    581     """Flush all current commands to the GLIR interpreter."""
--> 582     self._shared.flush(parser)
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>
        self._shared = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7a51c70>
        self = <vispy.gloo.glir.GlirQueue object at 0x2aacb7a51c40>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:504, in _GlirQueueShare.flush(self=<vispy.gloo.glir._GlirQueueShare object>, parser=<vispy.gloo.glir.GlirParser object>)
    502     show = self._verbose if isinstance(self._verbose, str) else None
    503     self.show(show)
--> 504 parser.parse(self._filter(self.clear(), parser))
        self = <vispy.gloo.glir._GlirQueueShare object at 0x2aacb7a51c70>
        parser = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:822, in GlirParser.parse(self=<vispy.gloo.glir.GlirParser object>, commands=[('FUNC', 'glClearColor', 0.0, 0.0, 0.0, 1.0), ('FUNC', 'glClear', 17664), ('SIZE', 39, 96), ('DATA', 39, 0, <class 'numpy.ndarray'> (8, 1) [('f0', '<f4', (3,))]), ('SIZE', 42, 56), ('DATA', 42, 0, <class 'numpy.ndarray'> (14,) uint32), ('FUNC', 'glDisable', 'cull_face'), ('FUNC', 'glDisable', 'depth_test'), ('FUNC', 'glEnable', 'blend'), ('FUNC', 'glBlendFuncSeparate', 'src_alpha', 'one_minus_src_alpha', 'zero', 'one'), ('DRAW', 41, 'triangle_strip', (42, 'UNSIGNED_INT', 14))])
    819     self._objects.pop(id_)
    821 for command in commands:
--> 822     self._parse(command)
        command = ('SIZE', 39, 96)
        self = <vispy.gloo.glir.GlirParser object at 0x2aacb7a51bb0>

File ~/.conda/envs/napari-env/lib/python3.9/site-packages/vispy/gloo/glir.py:778, in GlirParser._parse(self=<vispy.gloo.glir.GlirParser object>, command=('SIZE', 39, 96))
    776 if ob is None:
    777     if id_ not in self._invalid_objects:
--> 778         raise RuntimeError('Cannot %s object %i because it '
        'Cannot %s object %i because it '
                                       'does not exist' % (cmd, id_) = 'Cannot SIZE object 39 because it does not exist'
        cmd = 'SIZE'
        id_ = 39
        (cmd, id_) = ('SIZE', 39)
    779                            'does not exist' % (cmd, id_))
    780     return
    781 # Triage over command. Order of commands is set so most
    782 # common ones occur first.

RuntimeError: Cannot SIZE object 39 because it does not exist


```

On my laptop where I was doing this:

```
napari.manifest -> 'napari_stl_exporter:napari.yaml' could not be imported: Cannot find module 'napari_stl_exporter' declared in entrypoint: 'napari_stl_exporter:napari.yaml'
napari: 0.4.14
Platform: Linux-5.15.0-46-generic-x86_64-with-debian-bullseye-sid
System: Ubuntu 20.04.3 LTS
Python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37)  [GCC 9.3.0]
Qt: 5.15.1
PyQt5: 5.15.1
NumPy: 1.20.3
SciPy: 1.7.3
Dask: 2021.07.2
VisPy: 0.10.0

OpenGL:
  - GL version:  4.6 (Compatibility Profile) Mesa 21.2.6
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 1920x1080, scale 1.0

Plugins:
  - Visualization(B/C): 0.1.4
  - aicsimageio: 0.2.0
  - aicsimageio_delayed: 0.2.0
  - animation: 0.0.3.dev40+gaba975d
  - bioformats: 0.2.1
  - console: 0.0.4
  - itk-elastix_napari: 0.1.7
  - napari-cupy-image-processing: 0.1.4
  - napari-itk-io: 0.1.0
  - napari-medical-image-formats: 0.3.8
  - napari-nikon-nd2: 0.1.3
  - napari-stracking: 0.1.9
  - napari-tools-menu: 0.1.5
  - napari-tracks-reader: 0.1.3
  - napari_imagecodecs: 0.0.2
  - nd2-reader: 0.0.6
  - ome-types: 0.2.10
  - scikit-image: 0.4.14
  - svg: 0.1.5
torch.distributed.reduce_op is deprecated, please use torch.distributed.ReduceOp instead

```

In one machine right after conda --upgrade napari vispy

```
napari --info
napari.manifest -> 'napari-stl-exporter' could not be imported: Cannot find module 'napari_stl_exporter' declared in entrypoint: 'napari_stl_exporter:napari.yaml'
/home/jmamede/anaconda3/lib/python3.9/site-packages/superqt/qtcompat/__init__.py:7: UserWarning: The superqt.qtcompat module is deprecated as of v0.3.0. Please import from `qtpy` instead.
  warnings.warn(
napari: 0.4.14
Platform: Linux-5.15.0-43-generic-x86_64-with-glibc2.31
System: Ubuntu 20.04.4 LTS
Python: 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]
Qt: 5.15.2
PyQt5: 5.15.7
NumPy: 1.19.2
SciPy: 1.7.3
Dask: 2022.8.0
VisPy: 0.11.0

OpenGL:
  - GL version:  4.6.0 NVIDIA 510.60.02
  - MAX_TEXTURE_SIZE: 32768

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 1920x1080, scale 1.0

Plugins:
  - aicsimageio-in-memory: 0.6.1
  - aicsimageio-out-of-memory: 0.6.1
  - bfio: 2.3.0
  - bioformats: 0.2.1
  - console: 0.0.4
  - napari-elementary-numpy-operations: 0.0.5
  - napari-labelimg4classification: 0.1.1
  - napari-nikon-nd2: 0.1.3
  - napari-time-slicer: 0.4.9
  - napari-timeseries-opener-plugin: 0.1.2
  - napari-tools-menu: 0.1.11
  - napari_skimage_regionprops1: 0.5.3
  - napari_skimage_regionprops2: 0.5.3
  - nd2-reader: 0.0.6
  - ndtiffs: 0.1.2
  - ome-types: 0.2.10
  - scikit-image: 0.4.14
  - svg: 0.1.6

‼️  Errors were detected in the following plugins:
(Run 'napari --plugin-info -v' for more details)
  - lazy-openslide

```

--- a/src/napari/_vispy/canvas.py
+++ b/src/napari/_vispy/canvas.py
@@ -81,8 +81,19 @@ class NapariSceneCanvas(SceneCanvas_):
                 # hiding the actual source of the problem. They are never really
                 # informative, so we can safely ignore them.
                 pass
-            else:
-                raise
+            elif 'Shader compilation error' in error_msg:
+                raise RuntimeError(
+                    'Shader compilation failed. Unless you are working with custom shader code,\n'
+                    'this is likely a bug in napari.\n'
+                    'Please open an issue on the repository and provide this *full* stack trace.'
+                ) from e
+            elif 'Cannot SIZE object' in error_msg:
+                raise RuntimeError(
+                    'The above error may be caused by a version mismatch between vispy and napari.\n'
+                    'Try recreating a fresh environment and reinstalling. If that does not work,\n'
+                    'please open an issue on the repository and provide this *full* stack trace.'
+                ) from e
+            raise
 
 
 class VispyCanvas:

--- a/src/napari/_vispy/canvas.py
+++ b/src/napari/_vispy/canvas.py
@@ -80,14 +80,14 @@ class NapariSceneCanvas(SceneCanvas_):
                 # they flood the traceback because they are fired on each event,
                 # hiding the actual source of the problem. They are never really
                 # informative, so we can safely ignore them.
-                pass
-            elif 'Shader compilation error' in error_msg:
+                return
+            if 'Shader compilation error' in error_msg:
                 raise RuntimeError(
                     'Shader compilation failed. Unless you are working with custom shader code,\n'
                     'this is likely a bug in napari.\n'
                     'Please open an issue on the repository and provide this *full* stack trace.'
                 ) from e
-            elif 'Cannot SIZE object' in error_msg:
+            if 'Cannot SIZE object' in error_msg:
                 raise RuntimeError(
                     'The above error may be caused by a version mismatch between vispy and napari.\n'
                     'Try recreating a fresh environment and reinstalling. If that does not work,\n'

--- a/src/napari/_vispy/canvas.py
+++ b/src/napari/_vispy/canvas.py
@@ -66,6 +66,24 @@ class NapariSceneCanvas(SceneCanvas_):
             return
         super()._process_mouse_event(event)
 
+    def draw_visual(self, visual, event=None):
+        try:
+            super().draw_visual(visual, event=event)
+        except RuntimeError as e:
+            error_msg = e.args[0] if e.args else ''
+            to_ignore = (
+                'Cannot draw program if code has not been set',
+                'Cannot set uniform when program has no code',
+            )
+            if any(msg in error_msg for msg in to_ignore):
+                # these always happens after another (real) error is raised, and
+                # they flood the traceback because they are fired on each event,
+                # hiding the actual source of the problem. They are never really
+                # informative, so we can safely ignore them.
+                pass
+            else:
+                raise
+
 
 class VispyCanvas:
     """Class for our QtViewer class to interact with Vispy SceneCanvas. Also

